File: Data Structures & Algorithms/maximum-product-subarray/submission-3.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        mxProduct = nums[0]

        for i in range(len(nums)):
            curr = nums[i]
            mxProduct = max(mxProduct, curr)
            for j in range(i+1,len(nums)):
                curr *= nums[j]
                mxProduct = max(mxProduct,curr)
        
        return mxProduct

File: Data Structures & Algorithms/concatenation-of-array/submission-3.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n = nums.size();
        vector<int> arr(n*2, 0);
        for (int i=0;i<n;i++){
            arr[i] = nums[i];
            arr[n+i] = nums[i];
        }
        return arr;
    }
};
